backend/corpora/dataset_processing/upload.py

```diff
@@ -0,0 +1,139 @@
+            resp.raise_for_status()
+            with open(local_path, "wb") as fp:
+                for chunk in resp.iter_content(chunk_size=chunk_size):
+                    if tracker.stop_uploader.isSet():
+                        logger.debug("Upload ended early!")
+                        return
+                    elif chunk:
+                        fp.write(chunk)
+                        chunk_size = len(chunk)
+                        tracker.update(chunk_size)
+                        logger.debug(f"chunk size: {chunk_size}")
+    except requests.HTTPError as ex:
+        tracker.error.put(ex)
+    finally:
+        tracker.stop_updater.set()
+
+
+def processing_status_updater(uuid: str, updates: dict):
+    with db_session_manager() as manager:
+        manager.session.query(DbDatasetProcessingStatus).filter(DbDatasetProcessingStatus.id == uuid).update(updates)
+        manager.session.commit()
+
+
+def updater(upload_uuid: str, tracker: ProgressTracker, frequency: float = 3):
+    """
+    Update the progress of an upload to the database using the tracker.
+
+    :param upload_uuid: The uuid of the upload_progress row.
+    :param tracker: Tracks information about the progress of the upload.
+    :param frequency: The frequency in which the database is updated
+    :return:
+    """
+
+    def _update():
+        progress = tracker.progress()
+        if progress > 1:
+            tracker.stop_uploader.set()
+            processing_status = {
+                DbDatasetProcessingStatus.upload_progress: progress,
+                DbDatasetProcessingStatus.upload_message: "The file size, does not match the size of the upload.",
+                DbDatasetProcessingStatus.upload_status: UploadStatus.FAILED,
+            }
+        elif progress == 1 and tracker.stop_updater.isSet():
+            processing_status = {
+                DbDatasetProcessingStatus.upload_progress: progress,
+                DbDatasetProcessingStatus.upload_status: UploadStatus.UPLOADED,
+            }
+        else:
+            processing_status = {DbDatasetProcessingStatus.upload_progress: progress}
+        processing_status_updater(upload_uuid, processing_status)
+
+    try:
+        while not tracker.stop_updater.wait(frequency):
+            _update()
+        _update()  # Make sure the progress is update once the upload is complete
+    finally:
+        tracker.stop_uploader.set()
+
+
+def upload(dataset_uuid: str, url: str, local_path: str, file_size: int, chunk_size: int = 10 * MB, update_frequency=3):
+    """
+    Upload a file from a url and update the processing_status upload fields in the database
+
+    :param dataset_uuid: The uuid of the dataset the upload will be associated with.
+    :param url: The URL of the file to be uploaded.
+    :param local_path: The local name of the file be uploaded
+    :param file_size: The size of the file in bytes.
+    :param chunk_size: The size of downloaded data to copy to memory before saving to disk.
```

Comment:
can you add update_frequency to the param descriptions

Reply:
added